de4a-connector/src/main/java/eu/de4a/connector/api/manager/EvidenceTransferorManager.java

```diff
@@ -154,27 +156,34 @@ public void queueMessageResponse(MessageResponseOwner response) {
         }
     }
 
-    public boolean sendResponseMessage(String sender, String docTypeID, Element message, String tagContentId) {
+    public boolean sendResponseMessage(String receiverId, String dataOwnerId, String docTypeID, Element message, String tagContentId) {
         String errorMsg;
-        NodeInfo nodeInfo = client.getNodeInfo(sender, docTypeID, true,message);
         try {
-            String senderId = sender;
-            if(sender.contains(TCIdentifierFactory.PARTICIPANT_SCHEME + CIdentifier.URL_SCHEME_VALUE_SEPARATOR)) {
-                senderId = sender.replace(TCIdentifierFactory.PARTICIPANT_SCHEME + CIdentifier.URL_SCHEME_VALUE_SEPARATOR, "");
+            IParticipantIdentifier doPI = SimpleIdentifierFactory.INSTANCE
+                    .parseParticipantIdentifier(dataOwnerId.toLowerCase(Locale.ROOT));
+            IParticipantIdentifier rPI = SimpleIdentifierFactory.INSTANCE
+                    .parseParticipantIdentifier(receiverId.toLowerCase(Locale.ROOT));
+            if(doPI != null) {
+                dataOwnerId = doPI.getValue();
+            }
+            if(rPI != null) {
+                receiverId = rPI.getValue();
             }
             
+            NodeInfo nodeInfo = client.getNodeInfo(receiverId, docTypeID, true, message);
             DE4AKafkaClient.send(EErrorLevel.INFO, MessageFormat.format("Sending response message via AS4 gateway - "
                     + "DataEvaluatorId: {0}, Message tag: {1}, CanonicalEvidenceType: {2}", 
-                    senderId, tagContentId, docTypeID));
+                    receiverId, tagContentId, docTypeID));
```

Comment:
You are aware, that Message.format is very slow compared to String + String, right?

Reply:
Sure, but I think that it is insignificant, I dont think that our time leaks are here at all